lib/dock.rb

```diff
@@ -0,0 +1,54 @@
+class Dock
+  attr_reader     :name,
+                  :max_rental_time
+
+  def initialize(name, max_time)
+    @name            = name
+    @max_rental_time = max_time
+    @active_rentals  = []
+    @active_renters  = []
+    @return_boat     = []
+    @total_charges   = Hash.new(0)
+  end
+
+  def rent(boat, renter)
+    boat.rent_now
+    @active_rentals << boat
+    @active_renters << renter
+  end
+
+  def log_hour
+    @active_rentals.each do |rental|
+      rental.add_hour if rental.rented?
+    end
+  end
+
+  def return(boat)
+    if boat.hours_rented >= @max_rental_time
```

Comment:
This will only return a boat if it's rented for the max_rental_time or more, so a boat that is rented for less time would not generate revenue.